core/src/main/java/org/axonframework/eventsourcing/eventstore/jdbc/JdbcEventStorageEngine.java

```diff
@@ -296,7 +296,18 @@ protected void storeSnapshot(DomainEventMessage<?> snapshot, Serializer serializ
                                        PreparedStatement stmt = connection.prepareStatement(sql);
                                        stmt.setString(1, aggregateIdentifier);
                                        return stmt;
-                                   }, resultSet -> resultSet.next() ? resultSet.getObject(1, Long.class) : null,
+                                   },
+                                   		/* Fixed issue #636 */
+                   						// resultSet -> resultSet.next() ? resultSet.getObject(1, Long.class) : null
+                   						resultSet -> {
+                   							if (resultSet.next()) {
+                   								final Long value = resultSet.getObject(1, Long.class);
+                   								if (false == resultSet.wasNull()) {
+                   									return value;
+                   								}
+                   							}
+                   							return null;
+                   						},
```

Comment:
I agree with this approach. However, it would be much readable if this lambda is extracted into a method. That method can be reused later. I know I can benefit from it in #425.

Reply:
Or... we could make this method generic:
```
public static <T> T extract(ResultSet resultSet, int column, Class<T> columnType)
```
Btw, I like the idea of extracting it into a separate class. I'm not a big fan of `Utils` classes, maybe we can think of a better name (`ResultSetExtractor` e.g.)

Reply:
I like your idea.  What do you think of the following?

```java
public class ResultSetExtractor {

  public static <T> T nextAndExtract(ResultSet resultSet, int column, Class<T> columnType) throws SQLException {
    if (resultSet.next()) {
      return extract(resultSet, column, columnType);
    }

    return null;
  }

  public static <T> T extract(ResultSet resultSet, int column, Class<T> columnType) throws SQLException {
    final T value = resultSet.getObject(column, columnType);
    if (resultSet.wasNull()) {
      return null;
    }
  
    return value;
  }

  private ResultSetExtractor(){}
}
```

Reply:
Looks good!

Reply:
Will refactor it later on today.  Where should I put the new class (package)?

I would like to include a test class and was thinking of using mocks instead of a real database to simplify the tests.  What do you think?  If you agree, 
which mocking framework do you use?

Reply:
There is already `org.axonframework.common.jdbc.JdbcUtils` (what about that karma? :)) class which can hold these new methods.

We use mockito for mocking.

Reply:
Cheers.  Will make these changes later on and revert.

Reply:
Hi @javacreed,
given our target to release today, we'll be making these changes ourselves.
I will merge these changes so that you get the credits you deserve!
Thanks

Reply:
I am working on it now.  I will commit in a couple of minutes if you can wait.

Reply:
I am writing the test cases

Reply:
Great! I'll fix the build related stuff (setting up the container with MySQL).

Reply:
Cheers